b2sdk/_internal/scan/folder.py

```diff
@@ -245,6 +245,32 @@ def _walk_relative_paths(
             name = local_path.name
             relative_file_path = join_b2_path(relative_dir_path, name)
 
+            if policies_manager.exclude_all_symlinks and local_path.is_symlink():
+                if reporter is not None:
+                    reporter.symlink_skipped(str(local_path))
+                continue
+
+            if local_path.is_dir():
+                if policies_manager.should_exclude_local_directory(str(relative_file_path)):
+                    continue
+            else:
+                try:
+                    file_mod_time = get_file_mtime(str(local_path))
+                    file_size = local_path.stat().st_size
+                except OSError:
+                    # Skip broken symlinks or other inaccessible files
+                    file_mod_time = 0
+                    file_size = 0
+
+                local_scan_path = LocalPath(
+                    absolute_path=str(local_path.absolute()),
+                    relative_path=str(relative_file_path),
+                    mod_time=file_mod_time,
+                    size=file_size,
+                )
```

Comment:
your solution is valid, but please deduplicate this code


seems to me that `for (name, local_path, relative_file_path) in sorted(names):` is completely unneded as we could very well do sorting through `for local_path in sorted(local_dir.iterdir())`

that should make it easier to deplicate the localpath creation bit, but make your own decision here

Reply:
Yea, while fixing it I thought that this method needed some refactoring. Will work on it.

Reply:
@mjurbanski-reef did the dedup, any other suggestions?